nautilus_core/model/src/python/types/balance.rs

```diff
@@ -61,17 +63,32 @@ impl AccountBalance {
 
     #[staticmethod]
     #[pyo3(name = "from_dict")]
-    fn py_from_dict(py: Python<'_>, values: Py<PyDict>) -> PyResult<Self> {
-        from_dict_pyo3(py, values)
+    pub fn py_from_dict(py: Python<'_>, values: Py<PyDict>) -> PyResult<Self> {
+        let dict = values.as_ref(py);
+        let currency: &str = dict.get_item("currency")?.unwrap().extract()?;
+        let total_str: &str = dict.get_item("total")?.unwrap().extract()?;
+        let total: f64 = total_str.parse::<f64>().unwrap();
+        let free_str: &str = dict.get_item("free")?.unwrap().extract()?;
+        let free: f64 = free_str.parse::<f64>().unwrap();
+        let locked_str: &str = dict.get_item("locked")?.unwrap().extract()?;
+        let locked: f64 = locked_str.parse::<f64>().unwrap();
+        let currency = Currency::from_str(currency)?;
+        let account_balance = AccountBalance::new(
+            Money::new(total, currency)?,
+            Money::new(locked, currency)?,
+            Money::new(free, currency)?,
+        )
+        .unwrap();
+        Ok(account_balance)
     }
 
     #[pyo3(name = "to_dict")]
     pub fn py_to_dict(&self, py: Python<'_>) -> PyResult<PyObject> {
         let dict = PyDict::new(py);
         dict.set_item("type", stringify!(AccountBalance))?;
-        dict.set_item("total", self.total.to_string())?;
-        dict.set_item("locked", self.locked.to_string())?;
-        dict.set_item("free", self.free.to_string())?;
+        dict.set_item("total", self.total.as_f64().to_string())?;
```

Comment:
If we convert to floats, then we lose implicit precision with strings - which doesn't seem like a good idea?